doc/source/reference/random/index.rst

```diff
@@ -7,58 +7,146 @@
-distributions. See `NEP 19 <https://www.numpy.org/neps/
-nep-0019-rng-policy.html>`_ for context on the updated random Numpy number
-routines. The legacy `RandomState` random number routines are still
-available, but limited to a single BitGenerator. See :ref:`new-or-different` 
-for a complete list of improvements and differences from the legacy
-``RandomState``.
-
-For convenience and backward compatibility, a single `RandomState`
-instance's methods are imported into the numpy.random namespace, see
-:ref:`legacy` for the complete list.
-
 .. _random-quick-start:
 
 Quick Start
 -----------
 
-Call `default_rng` to get a new instance of a `Generator`, then call its
-methods to obtain samples from different distributions.  By default,
-`Generator` uses bits provided by `PCG64` which has better statistical
-properties than the legacy `MT19937` used in `RandomState`.
+The :mod:`numpy.random` module implements pseudo-random number generators
+(PRNGs) with the ability to draw samples from a variety of probability
+distributions. In general, users will create a `Generator` instance with
+`default_rng` and call the various methods on it to obtain samples from
+different distributions.
+
+::
+
+  >>> import numpy as np
+  >>> rng = np.random.default_rng()
+  # Generate one random float uniformly distributed over the range [0, 1)
+  >>> rng.random()  #doctest: +SKIP
+  0.06369197489564249  # may vary
+  # Generate an array of 10 numbers according to a unit Gaussian distribution.
+  >>> rng.standard_normal(10)  #doctest: +SKIP
+  array([-0.31018314, -1.8922078 , -0.3628523 , -0.63526532,  0.43181166,  # may vary
+          0.51640373,  1.25693945,  0.07779185,  0.84090247, -2.13406828])
+  # Generate an array of 5 integers uniformly over the range [0, 10).
+  >>> rng.integers(low=0, high=10, size=5)  #doctest: +SKIP
+  array([8, 7, 6, 2, 0])  # may vary
+
+PRNGs are deterministic sequences and can be reproduced by specifying a seed to
+control its initial state. By default, with no seed, `default_rng` will create
+the PRNG using nondeterministic data from the operating system and therefore
+generate different numbers each time. The pseudorandom sequences will be
+practically independent.
+
+::
+
+    >>> rng1 = np.random.default_rng()
+    >>> rng1.random()  #doctest: +SKIP
+    0.6596288841243357  # may vary
+    >>> rng2 = np.random.default_rng()
+    >>> rng2.random()  #doctest: +SKIP
+    0.11885628817151628  # may vary
+
+Seeds are usually large positive integers. `default_rng` can take positive
+integers of any size. We recommend using very large, unique numbers to ensure
+that your seed is different from anyone else's. This is good practice to ensure
+that your results are statistically independent from theirs unless if you are
+intentionally *trying* to reproduce their result. A convenient way to get
+such seed number is to use :py:func:`secrets.randbits` to get an
+arbitrary 128-bit integer.
```

Comment:
I consider `secrets.randbits(128)` to be the preferred idiom at this time. The `SeedSequence().entropy` spelling is not an API designed for this purpose. It's just an ad hoc way to hack around the existing implementation details to get what `default_rng(None)` would use.

Reply:
A bit unfortunate because the current doc on `SeedSequence` makes me think that this is the canonical way...

By the way I am also seeing that the doc of `Generator` is full of seeds like 1234 and 42.

Reply:
This is worth fixing in a follow-up, at least in the `numpy.random` docs themselves. Such seeding with short numbers is also done in very many other docstrings inside numpy and in other packages, but there it's a trade-off with being concise. I'd imagine those numbers could be made longer, but I would not expect all random number usages to use `secrets`.